### DISCUSSION TITLE
How cast empty string to null?

### DISCUSSION THREAD

**Discussion by cawa-93** [colinhacks/zod#1178] [Q&A]
I have optional text field. I need create validator for type `null` or `undefined` or _not empty string_.

I wrote:
```ts
const validator = string().trim().min(1).nullish()
```
In general, it works. But I would like the empty string to be interpreted as _no value_ and reduced to `null` or `undefined`

**Comment by StefanTerdell** [comment]
`z.string().transform(x => x || null)`
